### ISSUE TITLE
How do I test this?

### ISSUE THREAD

**Issue by timriker** [epacke/BigiIPReportFrontend#6]
Something like this would be very useful. Our current BigIPReport takes 2 minutes to load in a browser.

**Comment by timriker** [comment]
I know DataTables can take a json as source, but I don't know if we can talk it into rendering the tables, links and other things that are currently in the table. I'm not sure if DataTables would also render the entire table before displaying or not. A "react" front end or something similar would be awesome. load the json files into memory, do any searching and filtering from those data files, and dynamically render only the requested rows. It could leave out rendering the pool view until requested, and the status polling could handle dynamically adding or removing pool members as well as status.
Our current config has:
9 device groups
17 load balancers
6054 iRules
8597 pools
6996 virtualservers
10132 rows in BIG-IP Report
38704091 bytes in trim()ed "index.html"
Hmm. Might be nice to have a statistics page... I'll open an enhancement request. :)
The last reload 64 bit Chrome took 3.9 minutes before any of the json files were requested and then another 45 seconds after that to show the page. So we're almost up to 5 minutes to render the page.

**Comment by epacke** [comment]
> It could leave out rendering the pool view until requested, and the status polling could handle dynamically adding or removing pool members as well as status.

I don't think this particular thing is a good idea. Most people would not use BigIPReport if they had to create the polling nodes. From the sessions I've had with people around the globe almost nobody understands how it works. And setting it up can be painful depending on firewall rules and security policies. Besides, we'd lose the name property here (could potentially be looked up from the JSON though I suppose).

**Comment by timriker** [comment]
Oh, I don't mean removing the pool rendering, I mean that if you have polling setup, re-render the pool info so it gets the updates. The initial render would come from the json pool data.
